crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -66,12 +66,33 @@ pub trait StringNameSpaceImpl: AsString {
     // Parse a string number with base _radix_ into a decimal (i64)
     fn to_integer(&self, base: &UInt32Chunked, strict: bool) -> PolarsResult<Int64Chunked> {
         let ca = self.as_string();
-        let f = |opt_s: Option<&str>, opt_base: Option<u32>| -> Option<i64> {
+
+        let parse = |opt_s: Option<&str>, opt_base: Option<u32>| -> PolarsResult<i64> {
             match (opt_s, opt_base) {
-                (Some(s), Some(base)) => <i64 as Num>::from_str_radix(s, base).ok(),
-                _ => None,
+                (Some(s), Some(base)) if (2..=36).contains(&base) => {
+                    match <i64 as Num>::from_str_radix(s, base) {
+                        Ok(t) => Ok(t),
+                        Err(parse_int_error) => Err(PolarsError::ComputeError(
```

Comment:
In scala this would be `either.left.map(...)` but I don't know how to do that in Rust.